src/SFA.DAS.Commitments.Application/Commands/UpdateApprenticeshipStatus/PauseApprenticeshipCommandHandler.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Threading.Tasks;
+using FluentValidation;
+using MediatR;
+using SFA.DAS.Commitments.Application.Exceptions;
+using SFA.DAS.Commitments.Application.Services;
+using SFA.DAS.Commitments.Domain;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities;
+using SFA.DAS.Commitments.Domain.Entities.History;
+using SFA.DAS.Commitments.Domain.Interfaces;
+
+namespace SFA.DAS.Commitments.Application.Commands.UpdateApprenticeshipStatus
+{
+    public sealed class PauseApprenticeshipCommandHandler : AsyncRequestHandler<PauseApprenticeshipCommand>
+    {
+        private readonly IApprenticeshipRepository _apprenticeshipRepository;
+        private readonly ICommitmentRepository _commitmentRepository;
+        private readonly ICurrentDateTime _currentDate;
+        private readonly IApprenticeshipEvents _eventsApi;
+        private readonly IHistoryRepository _historyRepository;
+        private readonly ICommitmentsLogger _logger;
+        private readonly ApprenticeshipStatusChangeCommandValidator _validator;
+
+        public PauseApprenticeshipCommandHandler(
+            ICommitmentRepository commitmentRepository,
+            IApprenticeshipRepository apprenticeshipRepository,
+            ApprenticeshipStatusChangeCommandValidator validator,
+            ICurrentDateTime currentDate,
+            IApprenticeshipEvents eventsApi,
+            ICommitmentsLogger logger,
+            IHistoryRepository historyRepository
+        )
+        {
+            _commitmentRepository = commitmentRepository;
+            _apprenticeshipRepository = apprenticeshipRepository;
+            _validator = validator;
+            _currentDate = currentDate;
+            _eventsApi = eventsApi;
+            _logger = logger;
+            _historyRepository = historyRepository;
+        }
+
+        protected override async Task HandleCore(PauseApprenticeshipCommand command)
+        {
+            _logger.Info($"Employer: {command.AccountId} has called PauseApprenticeshipCommand", command.AccountId,
```

Comment:
The main code flow is basically the same. Is it worth templating the behaviour of the class?

Reply:
The handlers are already based on a class which would make it difficult to template. the existing base class is related to the MediatR operations and it is appearing to force separation to ensure the right message/handler binding works as intended.